clustering/jgroups/WFLY-8770_Integrate_aws_S3_PING_discovery_protocol.adoc

```diff
@@ -0,0 +1,114 @@
+= WFLY-8770 Integrate aws.S3_PING discovery protocol
+:author:            Radoslav Husar
+:email:             rhusar@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          jgroups,clustering,aws,ec2,discovery
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+`org.jgroups.protocols.aws.S3_PING` is a discovery protocol designed for JGroups cluster nodes running in AWS EC2.
+While there are other protocols that can be used in such an environment (`TCPPING`, `JDBC_PING`), the `org.jgroups.protocols.aws.S3_PING` is an ideal fit.
+Discovery is performed by writing and reading from an S3 bucket.
+Protocol is automatically configured from system properties.
+
+Historically all protocols were bundled within main `jgroups-<version>.jar`.
+Since the main JGroups project does not allow runtime dependencies, protocols had to include a lot of embedded unmaintainable copy-pasted and trimmed library code.
+This hinders supportability.
+Protocols needing to fetch dependencies now reside in separate repositories in the https://github.com/jgroups-extras organization.
+As such, `org.jgroups.aws.S3_PING` resides in the https://github.com/jgroups-extras/jgroups-aws repository.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/EAP7-863[EAP7-863] Integrate aws.S3_PING discovery protocol
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-8770[WFLY-8770] Integrate aws.S3_PING discovery protocol (upstream issue)
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:spriadka@redhat.com[Simon Priadka]
+
+=== Testing By
+
+* [ ] Engineering
+
+* [x] QE
+
+=== Affected Projects or Components
+
+* WildFly
+* JBoss EAP
+
+=== Other Interested Projects
+
+* Infinispan Server already bundles this protocol.
+
+=== Relevant Installation Types
+
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [ ] OpenShift s2i
+
+* [ ] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* `org.jgroups.protocols.aws.S3_PING` discovery protocol can be configured in the application server out of the box.
+* Protocol must be configured with `module="org.jgroups.aws"` attribute.
```

Comment:
this means the following:
```
<protocol type="S3_PING">
                        <property name="access_key">${jboss.jgroups.s3_ping.access_key}</property>
                        <property name="secret_access_key">${jboss.jgroups.s3_ping.secret_access_key}</property>
                        <property name="location">${jboss.jgroups.s3_ping.bucket}</property>
                    </protocol>
``` 
must become:
```
<protocol type="org.jgroups.protocols.aws.S3_PING" module="org.jgroups.aws">
                        <property name="access_key">${jboss.jgroups.s3_ping.access_key}</property>
                        <property name="secret_access_key">${jboss.jgroups.s3_ping.secret_access_key}</property>
                        <property name="location">${jboss.jgroups.s3_ping.bucket}</property>
                    </protocol>
```

would they both work? meaning the current setting would just be a sub-optimal configuration?

Reply:
@rhusar you already explained most of it in our chat:
"The EAP 7.4 uses JGroups 4.2.x which still contains the legacy S3 ping implementation, so that one remains available to the users. Users wanting to make use of the new implementation need to configure the protocol as we document, specifically:
```
<protocol type="org.jgroups.protocols.aws.S3_PING" module="org.jgroups.aws">                        <property name="region_name">${jboss.jgroups.aws.s3_ping.region_name}</property>                        <property name="bucket_name">${jboss.jgroups.aws.s3_ping.bucket_name}</property>                    </protocol>
```"

Reply:
it's not clear to me why we are willing to upgrade EAP7 since we won't lose the former functionality